route.go

```diff
@@ -437,6 +456,103 @@ func (nh *NextHop) decode(ad *netlink.AttributeDecoder) error {
 		}
 	}
 
+	if err := ad.Err(); err != nil {
+		return err
+	}
+
+	if encapType != 0 && encapBuf != nil {
+		// Found encapsulation, start decoding it from the buffer.
+		return nh.decodeEncap(encapType, encapBuf)
+	}
+
+	return nil
+}
+
+// An MPLSNextHop is a route next hop using MPLS encapsulation.
+type MPLSNextHop struct {
+	Label         int
+	TrafficClass  int
+	BottomOfStack bool
+	TTL           uint8
+}
+
+// TODO(mdlayher): MPLSNextHop TTL vs MPLS_IPTUNNEL_TTL. What's the difference?
+
+// encodeEncap encodes netlink attribute values related to encapsulation from
+// a NextHop.
+func (nh *NextHop) encodeEncap(ae *netlink.AttributeEncoder) error {
+	// TODO: this only handles MPLS encapsulation as that is all we support.
+
+	// Allocate enough space for an MPLS label stack.
+	var (
+		i int
+		b = make([]byte, 4*len(nh.MPLS))
```

Comment:
I feel like that `4` is used in a few places, it might be worth making it a const for mplsLabelSize

Reply:
4 bytes is actually the entire MPLS header. Calling it mplsLabelSize would be very inconsistent as the label is only a portion of those 4 bytes. If this is at all needed (which can be argued for clearity sake) I would call it mplsHeaderSize

Reply:
4 bytes is actually the entire MPLS header. Calling it mplsLabelSize would be very inconsistent as the label is only a portion of those 4 bytes. If this is at all needed (which can be argued for clearity sake) I would call it mplsHeaderSize